workerapi/api.go

```diff
@@ -98,9 +98,19 @@ func ThriftRunStatusToDomain(thrift *worker.RunStatus) runner.RunStatus {
 	if thrift.ExitCode != nil {
 		domain.ExitCode = int(*thrift.ExitCode)
 	}
+	if thrift.SnapshotId != nil {
+		domain.SnapshotID = *thrift.SnapshotId
+	}
 	return domain
 }
 
+func copyString(s string) *string {
```

Comment:
Are we being consistent elsewhere wrt nil and empty string when doing thrift <---> domain? Is this important other than the fact that it makes api_test.go easier?